TimeSeriesMetricsIT.testLongDimension failure

**Build scan**: https://gradle-enterprise.elastic.co/s/xukd3j37xp24i

**Repro line**:
```
gradlew ':server:internalClusterTest' --tests "org.elasticsearch.timeseries.support.TimeSeriesMetricsIT.testLongDimension" -Dtests.seed=4A1F56A5FD5F1AD6 -Dtests.locale=ar -Dtests.timezone=Etc/GMT-12 -Druntime.java=17
```

**Reproduces locally?**: Yes

**Applicable branches**: master

**Failure excerpt**:
```
org.elasticsearch.timeseries.support.TimeSeriesMetricsIT > testLongDimension FAILED
    java.lang.AssertionError: Expected a map containing
    Tuple [v1=v, v2={dim=0}]: a list containing
                             0: <2021-01-01T00:11:00.000Z=2.0>
                             1: expected <2021-01-01T00:15:00.000Z=3.0> but was <2021-01-01T00:20:00.000Z=4.0>
                             2: expected <2021-01-01T00:20:00.000Z=4.0> but was <missing>
    Tuple [v1=v, v2={dim=1}]: a list containing
                             0: <2021-01-01T00:11:00.000Z=5.0>
        at __randomizedtesting.SeedInfo.seed([4A1F56A5FD5F1AD6:D10DDEE0E69E8D67]:0)
        at org.elasticsearch.test.MapMatcher.assertMap(MapMatcher.java:82)
        at org.elasticsearch.test.MapMatcher.assertMap(MapMatcher.java:65)
        at org.elasticsearch.timeseries.support.TimeSeriesMetricsIT.assertSmallSimple(TimeSeriesMetricsIT.java:118)
        at org.elasticsearch.timeseries.support.TimeSeriesMetricsIT.testLongDimension(TimeSeriesMetricsIT.java:67)
```

--- a/server/src/internalClusterTest/java/org/elasticsearch/timeseries/support/TimeSeriesMetricsIT.java
+++ b/server/src/internalClusterTest/java/org/elasticsearch/timeseries/support/TimeSeriesMetricsIT.java
@@ -87,15 +87,11 @@ public class TimeSeriesMetricsIT extends ESIntegTestCase {
             "2021-01-01T00:20:00.000Z", };
         indexRandom(
             true,
-            client().prepareIndex("tsdb").setSource(Map.of("@timestamp", dates[0], "dim", d1, "v", 1)),
-            client().prepareIndex("tsdb").setSource(Map.of("@timestamp", dates[1], "dim", d1, "v", 2)),
-            client().prepareIndex("tsdb").setSource(Map.of("@timestamp", dates[2], "dim", d1, "v", 3)),
-            client().prepareIndex("tsdb").setSource(Map.of("@timestamp", dates[3], "dim", d1, "v", 4)),
-            client().prepareIndex("tsdb").setSource(Map.of("@timestamp", dates[1], "dim", d2, "v", 5, "m", 6)),
-            client().prepareIndex("tsdb").setSource(Map.of("@timestamp", dates[0], "dim", d1, "m", 1)),
-            client().prepareIndex("tsdb").setSource(Map.of("@timestamp", dates[1], "dim", d1, "m", 2)),
-            client().prepareIndex("tsdb").setSource(Map.of("@timestamp", dates[2], "dim", d1, "m", 3)),
-            client().prepareIndex("tsdb").setSource(Map.of("@timestamp", dates[3], "dim", d1, "m", 4))
+            client().prepareIndex("tsdb").setSource(Map.of("@timestamp", dates[0], "dim", d1, "v", 1, "m", 1)),
+            client().prepareIndex("tsdb").setSource(Map.of("@timestamp", dates[1], "dim", d1, "v", 2, "m", 2)),
+            client().prepareIndex("tsdb").setSource(Map.of("@timestamp", dates[2], "dim", d1, "v", 3, "m", 3)),
+            client().prepareIndex("tsdb").setSource(Map.of("@timestamp", dates[3], "dim", d1, "v", 4, "m", 4)),
+            client().prepareIndex("tsdb").setSource(Map.of("@timestamp", dates[1], "dim", d2, "v", 5, "m", 6))
         );
 
         assertMap(

